experimental/packages/otlp-exporter-base/src/types.ts

```diff
@@ -45,7 +45,7 @@ export interface ExportServiceError {
  * Collector Exporter base config
  */
 export interface OTLPExporterConfigBase {
-  headers?: Partial<Record<string, unknown>>;
+  headers?: Partial<Record<string, unknown> | (() => Record<string, string>)>;
```

Comment:
I think the `Partial` is unnecessary here as the `Record<>` type didn't add any required-key constraints. Also, it doesn't make sense to get a partial function object.

```suggestion
  headers?: Record<string, unknown> | () => Record<string, string>;
```

Reply:
Thanks for the tip, I've improved the config.headers type now